orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py

```diff
@@ -480,10 +488,21 @@ def testInstantiateORTTrainer(step_fn):
     model = pt_model.TransformerModel(28785, 200, 2, 200, 2, 0.2)
     my_loss = ort_utils.my_loss
     model_desc = ort_utils.transformer_model_description()
-    optim_config = optim.LambConfig()
-
+   
+    max_train_step = 1
+    warmup = 0.5
+    initial_lr = 1
+    optim_config = optim.SGDConfig() if not lr_scheduler else optim.SGDConfig(lr=initial_lr)
```

Comment:
@thiagocrepaldi when lr=1 is set but no lr_scheduler is given, the train step crashes by an input error. This is a temporary fix, but it's unclear what the problem is

Reply:
Can you repro this in the current API/example?

Reply:
Yes. I reproduced the error in two ways.

- Pass a learning rate of 1 at each train step
- Pass a learning rate scheduler to ORTTrainer that always returns 1

Error: 
`RuntimeError: Error in execution: Unexpected input data type. Actual: (N11onnxruntime17PrimitiveDataTypeIlEE) , expected: (N11onnxruntime17PrimitiveDataTypeIfEE)`

Reply:
At orttraining/orttraining/python/training/optim/config.py, around line 47, there is this:

`assert (isinstance(defaults['lr'], float) or isinstance(defaults['lr'], int)) and defaults['lr'] >= 0, "lr must be a positive number"`

Now that we create an internal IODescription with torch.float32, it is clear that we should change it to 

``assert isinstance(defaults['lr'], float) and defaults['lr'] >= 0, "lr must be a positive number"`` to prevent this issue

Reply:
@rayankrish